### DISCUSSION TITLE
Kickstart Switcher v2 - question

### DISCUSSION THREAD

**Discussion by RadioActive1983** [keirf/PCB-Projects#3] [General]
Hi.
Is there a possibility to use ST-Link V2 to program STM32 board with provided .hex file??

**Comment by keirf** [comment]
Yes though you may need a dfu file. I would need to add that as a target to the build I think.

**Comment by RadioActive1983** [comment]
Thanks for reply :)
OK, so I need some patience for it, right??
Second thing. I have and old Dell with WinXP and physical RS232 port.
Can I use it for configuration??

**Comment by keirf** [comment]
The STM32 serial port is TTL voltage levels. Really you need a USB serial TTL adapter. You could also use the same adapter for programming (using the HEX file).

**Comment by RadioActive1983** [comment]
Here comes an update.
I found  "HWK repair dongle" tool based on PL-2303HX and I'm pretty sure it's genuine one.
Next STM32 board connected with HWK dongle (5V straight from USB).
After that I installed Flash Loader Demo v2.8 and after running FLD.exe app recognises STM board.
Can I use FLD to write *hex file to device??
If yes, what about erasing process??
Set any additional options or just leave it as is and write hex??
Sorry for so many questions, but I don't want to destroy anything.

**Comment by keirf** [comment]
Yes that should work. I haven't used that application in a long time. If there are erase and/or verify options then use them. Apart from that, yes, just write the hex file.

**Comment by RadioActive1983** [comment]
Hello :)
One question.
I have STM32F103RCT6 based device.
Is there any chance, if I reprogram this with your .hex file, to make it working??

**Comment by keirf** [comment]
It will probably work, it just has larger Flash and SRAM. If it doesn't it might be because the Flash page size is different. Worth a try though.

**Comment by RadioActive1983** [comment]
So I tried.
Flash Loader Demo sent .hex into device with no errors, but when i went to reconfig step with putty there was only dark window, without any text, or reset button response.
Jumper between B4-B5 was set.
Is there any way to push it forward??

**Comment by keirf** [comment]
Unfortunately that can be all sorts of things. Not getting out of bootloader. Bad pins. Bad connection. Incompatible firmware is only one option here. I don't have one of these devices so I can't fix, assuming it is a firmware issue.

**Comment by keirf** [comment]
One thing it could be is a different speed crystal on your board. Can you see what is stamped on your crystal?

**Comment by RadioActive1983** [comment]
Stamp on crystal says 327A1D.

**Comment by keirf** [comment]
That's probably a low frequency 32.7kHz crystal. There should be a high frequency crystal too. Or take a photo or send a link to where you got the board. Should be able to work out the high frequency crystal type, assuming it has one

**Comment by RadioActive1983** [comment]
Internally this board has only one crystal, but this was easy to fix.
I took 8MHz crystal , capacitors, and resistor from a broken vacuum cleaner board, made tiny subboard for this set, and all of this "miracle of engineering" jumped into controller board.
Crystal from vacuum should be OK, faulty was power sector of motor driver.
Still dark screen on putty., no config list.

**Reply by keirf** [reply]
Unfortunately this is basically impossible to debug without time and access to the device.
